client/keystore/src/lib.rs

```diff
@@ -263,8 +283,13 @@ impl Store {
 				if let Some(name) = path.file_name().and_then(|n| n.to_str()) {
 					match hex::decode(name) {
 						Ok(ref hex) if hex.len() > 4 => {
-							if &hex[0..4] != &key_type.0 { continue }
-							let public = TPublic::from_slice(&hex[4..]);
+							let mut key_type: [u8; 4] = Default::default();
```

Comment:
Why do we have to make this copy now that we didn't do before to be able to compare the id?

Reply:
I reverted to using `&hex[0..4] != id.0` instead of copying.